packages/ui/src/components/vuestic-components/va-pagination/VaPagination.vue

```diff
@@ -1,215 +1,258 @@
-      },
-    },
-    disabled: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'disabled', false)
-      },
-    },
-    size: {
-      type: String,
-      default () {
-        return getContextPropValue(this, 'size', 'medium')
-      },
-      validator: value => {
-        return ['medium', 'small', 'large'].includes(value)
-      },
-    },
-    boundaryLinks: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'boundaryLinks', true)
-      },
-    },
-    directionLinks: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'directionLinks', true)
-      },
-    },
-    iconFont: {
-      type: Object,
-      default () {
-        return getContextPropValue(this, 'iconFont', null)
-      },
-    },
-    iconFontRight: {
-      type: Object,
-      default () {
-        return getContextPropValue(this, 'iconFontRight', null)
-      },
-    },
-  },
-  data () {
-    return {
-      defaultIconClass: {
-        direction: 'chevron_left',
-        boundary: 'first_page',
-      },
-      defaultIconRightClass: {
-        direction: 'chevron_right',
-        boundary: 'last_page',
-      },
+})
+export default class VaPagination extends mixins(...mixinsArr) {
+  private defaultIconClass: IconSet = {
+    direction: 'chevron_left',
+    boundary: 'first_page',
+  }
+
+  private defaultIconRightClass: IconSet = {
+    direction: 'chevron_right',
+    boundary: 'last_page',
+  }
+
+  private inputValue = ''
+
+  private useTotal = false
+
+  private get iconClass () {
+    return Object.assign({}, this.defaultIconClass, (this as any).iconSet)
+  }
+
+  private get iconRightClass () {
+    return Object.assign({}, this.defaultIconRightClass, (this as any).iconSetRight)
+  }
+
+  private get lastPage () {
+    const { total, pageSize, pages } = (this as any)
+    return this.useTotal
+      ? Math.ceil(total / pageSize)
+      : pages
+  }
+
+  private get paginationRange () {
+    const { valueComputed, visiblePages, total, pageSize, boundaryNumbers, pages } = (this as any)
+    return this.useTotal
+      ? setPaginationRange(valueComputed, visiblePages, Math.ceil(total / pageSize), boundaryNumbers)
+      : setPaginationRange(valueComputed, visiblePages, pages, boundaryNumbers)
+  }
+
+  private get showBoundaryLinks () {
+    const { visiblePages, boundaryLinks, boundaryNumbers } = (this as any)
+    return this.lastPage > visiblePages && boundaryLinks && !boundaryNumbers
+  }
+
+  private get showDirectionLinks () {
+    return this.lastPage > (this as any).visiblePages && (this as any).directionLinks
+  }
+
+  private get showPagination () {
+    return this.lastPage > 1 || (!(this as any).hideOnSinglePage && this.lastPage <= 1)
+  }
+
+  private get fontColor () {
+    return (this as any).computeColor((this as any).color)
+  }
+
+  @Watch('pageSize')
+  onPageSizeChange (newPageSize: number, oldPageSize: number) {
```

Comment:
This does nothing.

Reply:
It's a watcher for pageSize prop. And it looks like it's working :)

Reply:
Make as computed prop